webClient/src/app/shared/editor-control/editor-control.service.ts

```diff
@@ -583,6 +584,126 @@ export class EditorControlService implements ZLUX.IEditor, ZLUX.IEditorMultiBuff
     });
   }
 
+  public isContentValidForDataset(content: string[], datasetAttrs: DatasetAttributes): boolean | string {
```

Comment:
It looks like what this method really is checking for if the content is valid for a write operation. So I'd like to rename this to "isContentValidForDatasetWrite/DatasetSave" and why this is important because we don't check for 'U' undefined datasets (not that I see, at least) and we don't support those. So part of this method, we also want to check for undefined datasets. You can do this in two ways
1.  if maxRecordLen is 0
2. if recfm.recordLength is "U" (undefined)

If undefined dataset, `return 'Cannot save a dataset with Undefined record format'` or something like this... Functionally, this isn't a problem because currently we can't *open* undefined datasets in the Editor, so saving to it shouldn't be possible. But if for some reason we can attempt a save, then that will be a bug